### PR TITLE
Add wait queue RPC service

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

Needed by  https://github.com/OP-TEE/optee_os/pull/319

### CODE REVIEW

File: core/tee_wait_queue.c [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+#include <linux/slab.h>
+#include "tee_wait_queue.h"
+
```
Comment (jbech-linaro) [inline core/tee_wait_queue.c]: Also `#include <linux/types.h>` for `struct list_head` and `#include <linux/completion.h>` for `struct completion`.

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by jenswi-linaro** [comment]
Review tag applied

**Comment by jenswi-linaro** [comment]
Rebased and fixed travis checkpatch warning.

**Pull request merged**